tasks.c

```diff
@@ -424,8 +430,14 @@ PRIVILEGED_DATA static volatile TickType_t xPendedTicks = ( TickType_t ) 0U;
 PRIVILEGED_DATA static volatile BaseType_t xYieldPendings[ configNUMBER_OF_CORES ] = { pdFALSE };
 PRIVILEGED_DATA static volatile BaseType_t xNumOfOverflows = ( BaseType_t ) 0;
 PRIVILEGED_DATA static UBaseType_t uxTaskNumber = ( UBaseType_t ) 0U;
-PRIVILEGED_DATA static volatile TickType_t xNextTaskUnblockTime = ( TickType_t ) 0U;      /* Initialised to portMAX_DELAY before the scheduler starts. */
-PRIVILEGED_DATA static TaskHandle_t xIdleTaskHandles[ configNUMBER_OF_CORES ] = { NULL }; /*< Holds the handles of the idle tasks.  The idle tasks are created automatically when the scheduler is started. */
+PRIVILEGED_DATA static volatile TickType_t xNextTaskUnblockTime = ( TickType_t ) 0U; /* Initialised to portMAX_DELAY before the scheduler starts. */
+/* MISRA Ref 8.9.1 [Object in block scope] */
+/* More details at: https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/MISRA.md#rule-89 */
```

Comment:
xIdleTaskHandles is used in the following functions:
* xTaskGetIdleTaskHandle
* ulTaskGetIdleRunTimeCounter
* ulTaskGetIdleRunTimePercent
* prvCreateIdleTasks
It doesn't look like it should be defined in block scope.
Can you provide more information about this violation?

Reply:
I think it's also a false alarm. As you mentioned, it's referenced by several functions but it's noted to violate rule 8.9 by coverity.

> (1) Event misra_c_2012_rule_8_9_violation:	Symbol "xIdleTaskHandles" should be defined at block scope.
> Also see events:	[single_use]
> 436  	PRIVILEGED_DATA static TaskHandle_t xIdleTaskHandles[ configNUMBER_OF_CORES ] = /*< Holds the handles of the idle tasks.  The idle tasks are created automatically when the scheduler is started. */

> (2) Event single_use:	Symbol "xIdleTaskHandles" is only used in "xTaskGetIdleTaskHandle(void)".
Also see events:	[misra_c_2012_rule_8_9_violation]
3927 	    TaskHandle_t xTaskGetIdleTaskHandle( void )

Reply:
This alarm is because common code is suppressed by lint while FreeRTOS SMP is scanned by coverity. Suggest we describe the reason here and fixed when single core FreeRTOS is also scanned by coverity.